auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java

```diff
@@ -96,11 +104,29 @@ protected void checkedStart(Listener<RespT> responseListener, Metadata headers)
     };
   }
 
-  private Map<String, List<String>> getRequestMetadata() {
+  private URI serviceUri(Channel channel, MethodDescriptor<?, ?> method) throws StatusException {
+    String authority = channel.authority();
+    if (authority == null) {
+      throw Status.UNAUTHENTICATED.withDescription("Channel has no authority").asException();
+    }
+    final String suffixToStrip = ":443";
```

Comment:
I'm not too excited about that. DEFAULT_HTTPS_PORT isn't right, because it includes the colon. I could do `static final int DEFAULT_HTTPS_PORT = 443; ... String suffixToStrip = ":" + DEFAULT_HTTPS_PORT`, but I'm not sure who I'm helping.